frontend-react/src/components/StaticCompare.tsx

```diff
@@ -0,0 +1,143 @@
+import { ReactElement, useCallback, useEffect, useState } from "react";
+import DOMPurify from "dompurify";
+import { ScrollSync, ScrollSyncPane } from "react-scroll-sync";
+
+import { Diff, SES_TYPE } from "../utils/diff";
+import { splitOn } from "../utils/misc";
+
+interface StaticCompareProps {
+    rightText: string;
+    leftText: string;
+}
+
+const insertHighlight = (
+    s1: string,
+    offset: number,
+    length: number
+): string => {
+    if (s1 === "" || length === 0) {
+        return "";
+    }
+    // we want to insert a <span></span> around text.
+    const three_parts = splitOn(s1, offset, offset + length);
+    if (three_parts.length !== 3) {
+        console.error("split failed");
+        return s1;
+    }
+
+    return `${three_parts[0]}<mark>${three_parts[1]}</mark>${three_parts[2]}`;
+};
+
+export const StaticCompare = (props: StaticCompareProps): ReactElement => {
+    const [leftHighlightHtml, setLeftHighlightHtml] = useState("");
+    const [rightHighlightHtml, setRightHighlightHtml] = useState("");
+
+    const refreshHighlights = useCallback(
+        (rightText: string, leftText: string) => {
+            if (rightText.length === 0 || leftText.length === 0) {
+                return;
+            }
+
+            const differ = Diff(rightText, leftText);
+            differ.compose();
+            const sesses = differ.getses();
+
+            // because we're modifying text, it will change offsets UNLESS we go backwards.
+            // the later items in the patches array are sequential
+            let patchedLeftStr = rightText;
+            let patchedRightStr = leftText;
+
+            for (let ii = sesses.length - 1; ii >= 0; --ii) {
+                const eachses = sesses[ii];
```

Comment:
`eachSes`

Reply:
I like that name better. The names came the ported code.

Reply:
Done. FYI all this code is going away in my next PR.